### PR TITLE
add update_docs reusable workflow permission

### PR DESCRIPTION
**Pull request by juliannzhou**
This PR adds write permission to update_docs reusable workflow.

### CODE REVIEW

File: .github/workflows/update_docs.yml [hunk 1]
```diff
@@ -10,6 +10,8 @@ on:
     secrets:
       REPO_SCOPED_TOKEN:
         required: true
+    permissions:
```
Comment (oshi97) [inline .github/workflows/update_docs.yml:13]: out of curiosity when did this stop working?

Comment (juliannzhou) [reply .github/workflows/update_docs.yml:13]: it didn't work for my pr but it worked for a dummy pr that Yen created. Yen said that I should manually updated the doc in my pr and investigate later

Comment (oshi97) [reply .github/workflows/update_docs.yml:13]: do we know why it worked for Yen's dummy PR but not for your PR?

Comment (oshi97) [reply .github/workflows/update_docs.yml:13]: might be worth making an item for this

Comment (juliannzhou) [reply .github/workflows/update_docs.yml:13]: We never figured out why it failed on my PR on particular. After making some changes, it worked again.

Comment (oshi97) [reply .github/workflows/update_docs.yml:13]: Ok I'll close this for now but won't delete the branch!

**Pull request closed without merging**